### PR TITLE
Add min fee of .001 gas for any invocation

### PR DESCRIPTION
**Pull request by localhuman**
There is currently a network vulnerability that allows anyone to send invocation transactions that cost 9.9999 Gas worth of computation for free.

This could be used to spam the network and slow things down.

This pull request adds a minimum fee of .001 Gas for all contract invocations in order to prevent this.

### CODE REVIEW

**Comment by canesin** [comment]
I think this is great, but maybe it should be 0.0001, so that it is not free to attack but still very cheap to transact ?

**Comment by shargon** [comment]
I think lerider wanted to allow but not force

**Comment by erikzhang** [comment]
The `sys_fee` of any transaction should be a whole number, so it can't be 0.001.
But if we set the `min_gas` to 1, it is too high.

**Comment by canesin** [comment]
@erikzhang where tx fees (the ones that enter in NetworkFee) goes ? If it is redistributed to NEO holders I think is better. Groups in control of consensus nodes will have many NEO already, so they benefit also.

**Comment by erikzhang** [comment]
SystemFee and the block award is redistributed to NEO holders. And NetworkFee is redistributed to consensus nodes.

**Comment by canesin** [comment]
yes @erikzhang , my question is: shouldn't that be change so that NetworkFee also is redistributed to NEO holders ? I see no scenario that consensus nodes are not somehow holding large amounts of NEO in its influence network.

**Comment by erikzhang** [comment]
There will be such a scenario soon.

File: neo/neo.csproj [hunk 1]
```diff
@@ -37,17 +37,18 @@
...
 
   <ItemGroup Condition="'$(TargetFramework)' == 'net47'">
-    <PackageReference Include="Replicon.Cryptography.SCrypt">
-      <Version>1.1.6.13</Version>
+    <PackageReference Include="Replicon.Cryptography.SCrypt" Version="1.1.6.13">
     </PackageReference>
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="..\..\neo-vm\src\neo-vm\neo-vm.csproj" />
```
Comment (shargon) [inline neo/neo.csproj]: Be carefull with this

**Pull request closed without merging**